test/lit.cfg

```diff
@@ -1998,3 +1998,18 @@ lit_config.note("Available features: " + ", ".join(sorted(config.available_featu
 # Windows currently does not support basic reflection support.
 if 'objc_interop' in config.available_features or ('swift_stdlib_asserts' in config.available_features and not kIsWindows):
     config.available_features.add('reflection_test_support')
+
+# num_extra_inhabitants on 64 bits differs between mac and everywhere else.
+# Add it, as well as the the 3 previous values, as variables,
+# in order to substitute them in the tests
+if platform.system() == 'Darwin':
+    num_extra_inhabitants_64bit = 2147483647
+else:
+    num_extra_inhabitants_64bit = 4096
+add_num_extra_inhabitants = "-Dnum_extra_inhabitants_64bit={} ".format(num_extra_inhabitants_64bit)
+
+for i, num in enumerate(range(num_extra_inhabitants_64bit - 1, num_extra_inhabitants_64bit - 4, -1)):
```

Comment:
It's possible to specify numeric variables in FileCheck now using "-D#foo=val", see https://llvm.org/docs/CommandGuide/FileCheck.html#filecheck-numeric-substitution-blocks. I think that would be a good fit here, since it'd allow us to just introduce one predefined value. For example we could write "-D#num_extra_inhabitants_64bit=4096" in lit.cfg, then write "[[#num_extra_inhabitants_64bit]]", "[[#num_extra_inhabitants_64bit-1]]", etc. in tests.

Reply:
It looks way cleaner now! Just waiting for the current tests to end before pushing.